### PR TITLE
Don't treat initialization to 'false' as assignment.

### LINKED ISSUE #35: inconsistent behavior with bool initialization
This tool has allowed me to slay many a bug today. Thank you!!!

I also noticed this inconsistent behavior.

```
package main

import "fmt"

func main() {

	s := ""
	s = "hello"
	fmt.Println("s:", s)

	n := 0
	n = 42
	fmt.Println("n:", n)

	b := false
	b = true
	fmt.Println("b:", b)
}
```

I can init and update `s` and `n` as expected, but I'm getting `ineffectual assignment to b`.

I am running go1.12.5.
 
Thanks!

### PR DESCRIPTION
**Pull request by jvmatl**
Fixes #35

### CODE REVIEW

**Comment by jvmatl** [comment]
After I submitted this, I realized there's a tiny flaw: *go does not treat `false` as a reserved word* -- personally, I'm ok with ignoring this, because anybody redefines `false` deserves as much bad karma as the world can throw at them.

https://play.golang.org/p/1HeKi6j3zVd

File: ineffassign.go [hunk 1]
```diff
@@ -368,14 +368,17 @@ func isZeroInitializer(x ast.Expr) bool {
...
 	}
 
-	b, ok := x.(*ast.BasicLit)
-	if !ok {
-		return false
-	}
-	switch b.Value {
-	case "0", "0.0", "0.", ".0", `""`:
-		return true
+	switch x.(type) {
```
Comment (gordonklaus) [inline ineffassign.go]: Write this as `switch x := x.(type) {` and then you won't have to repeat the type assertions below.

Comment (jvmatl) [reply ineffassign.go]: TIL! I didn't know you could do that. Looks much better.

Is this what you had in mind?

Comment (gordonklaus) [reply ineffassign.go]: Precisely 😄

**Review by Scutua — APPROVED** [review]

**Pull request merged**